Use native accessor for handler map interactions

nickiaconis opened the pull request:
Most interactions with `_handlerMap` already use native accessors (dots and square brackets). However, the one in `_getTargetEventHandler` uses `Ember.get`.

This creates an issue for class selector strings as they begin with a dot (e.g. `.actor-list`). In order to use `Ember.Object`'s getter, the target is included in a property path string. Property path strings use dots to delimit properties, so the path becomes, for example, `_handlerMap..actor-list.click`.

This causes an error to be thrown in recent versions of Ember, which assert that `Ember.get` is not used with an empty string as the property name. The `..` substring is interpreted by `Ember.get` to delimit an additional property name on the property path, which has a length of zero, i.e. an empty string.

Additionally, it stands to reason that any selector that contains a class would produce an unexpected result. Take `unifiedEventHandler.register('a.foo', 'click', onClick);` for instance. This will cause the property path `_handlerMap.a.foo.click` to be looked up on the service, which is akin to `this['_handlerMap']['a']['foo']['click']`. However, any existing handler will have been stored at `this['_handlerMap']['a.foo']['click']`, leading the property lookup via `Ember.get` to return `undefined` instead of the existing handler.

N.B. This PR doesn't fix the lack of account for different selectors targeting the same element.

trentmwillis commented on addon/services/unified-event-handler.js:
```diff
@@ -122,8 +122,8 @@ export default Ember.Service.extend(Ember.Evented, {
-        this[_HANDLER_MAP][target] = Object.create(null);
-        this[_HANDLER_MAP][target][eventName] = handlerInfo;
+        handlerMap[target] = targetHandlers = Object.create(null);
+        targetHandlers[eventName] = handlerInfo;
```
Since this is the same as the `else` block below. Can we eliminate the `else` and just have this on one line?

nickiaconis replied:
Good point; fixed!

nickiaconis commented:
Hey @trentmwillis, thanks for the review! I've added a test (or, rather, modified an existing one) for coverage. I pushed the commits separately to demonstrate the bug and show that my changes fix it.
